graylog2-server/src/main/java/org/graylog2/entitygroups/rest/EntityGroupResource.java

```diff
@@ -0,0 +1,202 @@
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import jakarta.ws.rs.BadRequestException;
+import jakarta.ws.rs.ForbiddenException;
+import jakarta.ws.rs.NotFoundException;
+import org.apache.shiro.authz.annotation.RequiresAuthentication;
+import org.apache.shiro.authz.annotation.RequiresPermissions;
+
+import org.graylog2.audit.AuditEventTypes;
+import org.graylog2.audit.jersey.AuditEvent;
+import org.graylog2.audit.jersey.NoAuditEvent;
+import org.graylog2.entitygroups.EntityGroupService;
+import org.graylog2.entitygroups.model.EntityGroup;
+import org.graylog2.database.PaginatedList;
+import org.graylog2.plugin.rest.PluginRestResource;
+import org.graylog2.rest.models.SortOrder;
+import org.graylog2.shared.rest.resources.RestResource;
+
+import jakarta.inject.Inject;
+
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.DefaultValue;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.graylog2.shared.utilities.StringUtils;
+
+import java.util.function.Predicate;
+
+import static org.graylog2.entitygroups.rest.EntityGroupPermissions.ENTITY_GROUP_CREATE;
+import static org.graylog2.entitygroups.rest.EntityGroupPermissions.ENTITY_GROUP_DELETE;
+import static org.graylog2.entitygroups.rest.EntityGroupPermissions.ENTITY_GROUP_EDIT;
+import static org.graylog2.entitygroups.rest.EntityGroupPermissions.ENTITY_GROUP_READ;
+import static org.graylog2.shared.utilities.ExceptionUtils.getRootCauseMessage;
+
+@Api(value = "EntityGroups", description = "Manage Entity Groups")
+@Path("/entity_groups")
+@RequiresAuthentication
+@Produces(MediaType.APPLICATION_JSON)
+public class EntityGroupResource extends RestResource implements PluginRestResource {
+
+    private final EntityGroupService entityGroupService;
+
+    @Inject
+    public EntityGroupResource(EntityGroupService entityGroupService) {
+        this.entityGroupService = entityGroupService;
+    }
+
+    @GET
+    @ApiOperation(value = "Get a list of entity groups")
+    public PaginatedList<EntityGroup> listGroups(@ApiParam(name = "page") @QueryParam("page") @DefaultValue("1") int page,
+                                                 @ApiParam(name = "per_page") @QueryParam("per_page") @DefaultValue("15") int perPage,
+                                                 @ApiParam(name = "query") @QueryParam("query") @DefaultValue("") String query,
+                                                 @ApiParam(name = "sort",
+                                                           value = "The field to sort the result on",
+                                                           allowableValues = "name")
```

Comment:
Nit: Similar to some other pagination endpoints, could we add a default value in the pagination endpoints to avoid an error when it's not provided? 
```suggestion
                                                           allowableValues = "name")
                                                 @DefaultValue(EntityGroup.FIELD_NAME)
```